src/blockcrawler/evm/rpc.py

```diff
@@ -195,6 +195,77 @@ async def get_block(self, block_num: HexInt, full_transactions: bool = False) ->
         )
         return block
 
+    async def get_block_by_timestamp(self, timestamp: int) -> int:
+        """Get block number by timestamp.
+        `Source <https://ethereum.stackexchange.com/a/127720>`_
```

Comment:
This doesn't help the user understand the function. Please add it as a separate comment to acknowledge that it was derived from the stack exchange article.